plugins/obs-ffmpeg/texture-amf.cpp

```diff
@@ -1161,16 +1161,38 @@ static void check_texture_encode_capability(obs_encoder_t *encoder,
 
 #include "texture-amf-opts.hpp"
 
-static void amf_defaults(obs_data_t *settings)
+static void amf_avc_defaults(obs_data_t *settings)
 {
+	// These are initial recommended settings that may be lowered later
+	// once we know more info such as the resolution and frame rate.
+	obs_data_set_default_string(settings, "rate_control", "CBR");
 	obs_data_set_default_int(settings, "bitrate", 2500);
 	obs_data_set_default_int(settings, "cqp", 20);
-	obs_data_set_default_string(settings, "rate_control", "CBR");
 	obs_data_set_default_string(settings, "preset", "quality");
 	obs_data_set_default_string(settings, "profile", "high");
 	obs_data_set_default_int(settings, "bf", 3);
 }
 
+static void amf_hevc_defaults(obs_data_t *settings)
+{
+	// These are initial recommended settings that may be lowered later
+	// once we know more info such as the resolution and frame rate.
+	obs_data_set_default_string(settings, "rate_control", "VBR");
```

Comment:
VBR offers better quality than CBR for video-on-demand content such as when recording where bits can be allocated according to the need/complexity of each scene. With the current code, VBR would apply for OBS recordings.

CBR is more suited for real time live applications such as streaming where OBS
would override the rate control method to a hard coded CBR when streaming anyways:

https://github.com/obsproject/obs-studio/blob/master/plugins/rtmp-services/rtmp-common.c#L665
```obs_data_set_string(settings, "rate_control", "CBR");```

Reply:
While OBS currently overrides the rate control method to CBR for streaming, it may not always in the future. I prefer that all rate control methods are the same default and that users learn to pick the rate control that is most appropriate for their use case.

Reply:
Updated RATE_CONTROL_METHOD default to CBR